### PR TITLE
Add MaxRequeueNum as a controller-manager param

### LINKED ISSUE #1031: Add maxRetries parameters to controller
/kind feature

**What happened**:

```
const (
	// maxRetries is the number of times a volcano job will be retried before it is dropped out of the queue.
	// With the current rate-limiter in use (5ms*2^(maxRetries-1)) the following numbers represent the times
	// a volcano job is going to be requeued:
	//
	// 5ms, 10ms, 20ms, 40ms, 80ms, 160ms, 320ms, 640ms, 1.3s, 2.6s, 5.1s, 10.2s, 20.4s, 41s, 82s
	maxRetries = 15
)
```

the maxRetries is hard coded.

And In my scenario, I hope the job controller keep retrying, just like other k8s controllers.

So, is it ok to add  a maxRetries parameter, default value is 15, when set to -1, it means retry endlessly

**Environment**:
- Volcano Version: v1.0.1

### PR DESCRIPTION
**Pull request by shinytang6**
original #1078, resolves #1031
Signed-off-by: tangliang <1074461480@qq.com>

### CODE REVIEW

File: pkg/controllers/job/job_controller.go [hunk 1]
```diff
@@ -139,6 +131,10 @@ func (cc *jobcontroller) Initialize(opt *framework.ControllerOption) error {
 	cc.errTasks = newRateLimitingQueue()
 	cc.recorder = recorder
 	cc.workers = workers
+	cc.maxRequeueNum = opt.MaxRequeueNum
+	if cc.maxRequeueNum < 0 {
```
Comment (k82cn) [inline pkg/controllers/job/job_controller.go:135]: please check this parameter to make sure it is larger than zero :)

File: pkg/controllers/job/job_controller.go [hunk 2]
```diff
@@ -328,7 +324,7 @@ func (cc *jobcontroller) processNextReq(count uint32) bool {
 	}
 
 	if err := st.Execute(action); err != nil {
-		if queue.NumRequeues(req) < maxRetries {
+		if cc.maxRequeueNum == -1 || queue.NumRequeues(req) < cc.maxRequeueNum {
```
Comment (k82cn) [inline pkg/controllers/job/job_controller.go:327]: please update `-1` checking to `cc.maxRequeueNum  < 0`

File: pkg/controllers/queue/queue_controller.go [hunk 3]
```diff
@@ -254,7 +251,7 @@ func (c *queuecontroller) handleQueueErr(err error, obj interface{}) {
 		return
 	}
 
-	if c.queue.NumRequeues(obj) < maxRetries {
+	if c.maxRequeueNum == -1 || c.queue.NumRequeues(obj) < c.maxRequeueNum {
```
Comment (k82cn) [inline pkg/controllers/queue/queue_controller.go:254]: ditto

File: pkg/controllers/queue/queue_controller.go [hunk 4]
```diff
@@ -323,7 +320,7 @@ func (c *queuecontroller) handleCommandErr(err error, obj interface{}) {
 		return
 	}
 
-	if c.commandQueue.NumRequeues(obj) < maxRetries {
+	if c.maxRequeueNum == -1 || c.commandQueue.NumRequeues(obj) < c.maxRequeueNum {
```
Comment (k82cn) [inline pkg/controllers/queue/queue_controller.go:323]: ditto

File: pkg/controllers/job/job_controller.go (hunk 1 above)
Comment (shinytang6) [reply pkg/controllers/job/job_controller.go:135]: Not sure if I fully understand what you mean. Do you mean `cc.maxRequeueNum` can't be negative? l think the controller should retry forever when `MaxRequeueNum` is set to negative

File: cmd/controller-manager/app/options/options.go [hunk 5]
```diff
@@ -40,6 +41,11 @@ type ServerOption struct {
 	// WorkerThreads is the number of threads syncing job operations
 	// concurrently. Larger number = faster job updating, but more CPU load.
 	WorkerThreads uint32
+	// MaxRequeueNum is the number of times a job, queue or command will be requeued before it is dropped out of the queue.
+	// With the current rate-limiter in use (5ms*2^(maxRetries-1)) the following numbers represent the times
+	// a job, queue or command is going to be requeued:
+	// 5ms, 10ms, 20ms, 40ms, 80ms, 160ms, 320ms, 640ms, 1.3s, 2.6s, 5.1s, 10.2s, 20.4s, 41s, 82s
+	MaxRequeueNum int
```
Comment (Thor-wl) [inline cmd/controller-manager/app/options/options.go:48]: Please add some usage introduction and examples in design doc

**Pull request merged**